### PR TITLE
[SYCL][Bindless] Fix get_image_num_channels failing with mipmap images

### PR DESCRIPTION
**Pull request by DBDuncan**
This patch fixes a crash due to get_image_num_channels expecting a CUarray instead of a CUmipmappedArray. Expands test coverage to test this.

### CODE REVIEW

**Comment by DBDuncan** [comment]
Corresponding UR PR: https://github.com/oneapi-src/unified-runtime/pull/1961

File: sycl/cmake/modules/FetchUnifiedRuntime.cmake [hunk 1]
```diff
@@ -115,14 +115,14 @@ if(SYCL_PI_UR_USE_FETCH_CONTENT)
       CACHE PATH "Path to external '${name}' adapter source dir" FORCE)
   endfunction()
 
-  set(UNIFIED_RUNTIME_REPO "https://github.com/oneapi-src/unified-runtime.git")
+  set(UNIFIED_RUNTIME_REPO "git@git.office.codeplay.com:sycl/unified-runtime.git")
```
Comment (cppchedy) [inline sycl/cmake/modules/FetchUnifiedRuntime.cmake:118]: reset this to `https://github.com/oneapi-src/unified-runtime.git"`.

Comment (DBDuncan) [reply sycl/cmake/modules/FetchUnifiedRuntime.cmake]: Fixed. Updated to `https://github.com/DBDuncan/unified-runtime.git` for CI testing.

**Review by cppchedy — APPROVED** [review]

**Review by omarahmed1111 — APPROVED** [review]

**Comment by omarahmed1111** [comment]
@intel/llvm-gatekeepers Please merge, Thanks!

File: sycl/cmake/modules/FetchUnifiedRuntime.cmake [hunk 2]
```diff
@@ -117,13 +117,13 @@ if(SYCL_UR_USE_FETCH_CONTENT)
...
-  # commit b4195cbb694b6e7c6216bf42a757ed700a325855
-  # Merge: 5482621c ab77d020
+  # commit 76361a88a0af6502fe655144655b381260ea9706
+  # Merge: b4195cb 0db57da
   # Author: Omar Ahmed <omar.ahmed@codeplay.com>
-  # Date:   Mon Aug 26 14:00:04 2024 +0100
-  #    Merge pull request #2006 from zhaomaosu/devsan-fix-cxxlib-mismatch
-  #    [DeviceSanitizer] Fix libstdc++ and libc++ mismatch problem
-  set(UNIFIED_RUNTIME_TAG b4195cbb694b6e7c6216bf42a757ed700a325855)
+  # Date:   Wed, 31 Jul 2024 14:26:38 +0100
```
Comment (AlexeySachkov) [inline sycl/cmake/modules/FetchUnifiedRuntime.cmake:123]: @omarahmed1111, I somehow found this commit and I wonder: did we really reverted a UR version to a commit which is a *month* older than what we used to have?

How did it not cause any regressions? Or am I missing something here?

Comment (kbenzie) [reply sycl/cmake/modules/FetchUnifiedRuntime.cmake:123]: Looks like the date in the comment was incorrect but the rest of the message and commit hash were correct.

```
commit 76361a88a0af6502fe655144655b381260ea9706
Merge: b4195cbb 0db57da9
Author: Omar Ahmed <omar.ahmed@codeplay.com>
Date:   Mon Aug 26 15:57:45 2024 +0100

    Merge pull request #1961 from DBDuncan/duncan/num-channels-mipmap-fix

    [Bindless][Exp] Fix urBindlessImagesImageGetInfoExp failing with mipmap images
```

Comment (AlexeySachkov) [reply sycl/cmake/modules/FetchUnifiedRuntime.cmake:123]: Thanks for the clarification, @kbenzie!

**Pull request merged**